pkg/cli/proxy_get.go

```diff
@@ -65,9 +65,11 @@ func GetEnvoyProxyConfig(clientSet kubernetes.Interface, config *rest.Config, na
 	return envoyProxyConfig, nil
 }
 
-// proxyLabelExists returns a boolean indicating if the pod is part of a mesh
-func proxyLabelExists(pod corev1.Pod) bool {
-	// osm-controller adds a unique label to each pod that belongs to a mesh
+// ProxyLabelExists returns a boolean indicating if the pod is part of a mesh.
+// Note that this is NOT a definitive check indicating pod membership in a mesh
+// as the label may continue to exist even if the pod is removed from the mesh.
+func ProxyLabelExists(pod corev1.Pod) bool {
+	// osm-controller adds a unique label to a pod when it is added to a mesh
 	proxyUUID, proxyLabelSet := pod.Labels[constants.EnvoyUniqueIDLabelName]
 	return proxyLabelSet && isValidUUID(proxyUUID)
 }
```

Comment:
These sort of generic utility functions can live in `pkg/mesh`. It shouldn't have to be exported from the CLI pkg if it's meant to be reused outside the CLI. `pkg/cli` pertains to osm CLI utilities.

Reply:
@shashankram Moved to `pkg/mesh`